apps/schema2/src/components/ColumnEditModal.vue

```diff
@@ -0,0 +1,239 @@
+            class="col-4"
+            v-if="
+              column.columnType == 'REFBACK' &&
+              refBackCandidates(column.refTable, table.name).length > 1
+            "
+          >
+            <InputSelect
+              id="column_refBack"
+              label="refBack"
+              v-model="column.refBack"
+              :options="refBackCandidates(column.refTable, table.name)"
+            />
+          </div>
+          <div
+            class="col-4"
+            v-if="
+              column.refTable &&
+              (column.columnType == 'REF' || column.columnType == 'REF_ARRAY')
+            "
+          >
+            <InputString
+              id="column_refLink"
+              v-model="column.refLink"
+              label="refLink"
+            />
+          </div>
+        </div>
+        <div class="d-flex align-content-around flex-wrap">
+          <div class="col-4" v-if="column.columnType != 'CONSTANT'">
+            <InputText
+              id="column_validation"
+              v-model="column.validation"
+              label="validationExpression"
+              description="Example: {name} == 'John'"
+            />
+          </div>
+          <div class="col-4">
+            <InputText
+              id="column_visible"
+              v-model="column.visible"
+              label="visibleIf"
+              description="{other} > 5"
+            />
+          </div>
+          <div class="col-4">
+            <InputString
+              id="column_semantics"
+              v-model="column.semantics"
+              :list="true"
+              label="semantics"
+            />
+          </div>
+        </div>
+      </LayoutForm>
+    </template>
+    <template v-slot:footer>
+      <ButtonAction @click="close">Done</ButtonAction>
+      <ButtonDanger @click="toggleDelete" v-if="column.drop != true">
+        Mark as deleted
+      </ButtonDanger>
+      <ButtonDanger @click="toggleDelete" v-else>
+        Undo mark as deleted
+      </ButtonDanger>
+    </template>
+  </LayoutModal>
+</template>
+
+<script>
+import {
+  LayoutForm,
+  InputText,
+  InputString,
+  InputBoolean,
+  InputSelect,
+  IconAction,
+  LayoutModal,
+  ButtonAction,
+  ButtonDanger,
+  MessageWarning,
+} from "molgenis-components";
+import columnTypes from "../columnTypes.js";
+
+export default {
+  components: {
+    LayoutForm,
+    InputText,
+    InputString,
+    InputBoolean,
+    InputSelect,
+    IconAction,
+    LayoutModal,
+    ButtonAction,
+    ButtonDanger,
+    MessageWarning,
+  },
+  props: {
+    /** Column metadata object entered as v-model */
+    value: Object,
+    /** table that column is part of */
+    table: Object,
+    /** schema  column is part of */
+    schema: Object,
+  },
+  data() {
+    return {
+      //show
+      show: false,
+      // of type 'column metadata'
+      column: null,
+      //the options
+      columnTypes,
+      //for forcing updates
+      timestamp: Date.now(),
```

Comment:
this will break ssr compatibility due to dehydration mismatch

Reply:
tried to remove silly timestamps. Should have been done by now